script.js

```diff
@@ -1,4 +1,159 @@
+// -- Beat That! -- //
+
+// 1. Two players take turn
+// -> change of 'mode' for different players
+
+// 2. After clicking submit, the game rolls 2 dice and display the dice rolls
+// -> global variables needed to store the dice values for each player
+
+// 3. The player can choose the order they want to display the dice (e.g. specific 2nd dice go first)
+// -> conditional statements (i.e. if choose 1...)
+
+// 4. Players with the higher number between 2 players win the game
+// -> needs a comparison function, to compare the absolute value (number)
+
+//-----------------------------------
+//       Global Variables
+//-----------------------------------
+
+var player1diceRoll1 = "";
+var player1diceRoll2 = "";
+var player2diceRoll1 = "";
+var player2diceRoll2 = "";
+var playerOneChoice = [];
+var playerTwoChoice = [];
+var typeOneCombinedForPlayer1 = "";
+var typeTwoCombinedForPlayer1 = "";
+var typeOneCombinedForPlayer2 = "";
+var typeTwoCombinedForPlayer2 = "";
+var numOfPlayer1Wins = 0;
+var numOfPlayer2Wins = 0;
+var currentGameMode = "waiting for player 1";
+var myOutputValue = "";
+
+//-----------------------------------
+//       Helper Functions
+//-----------------------------------
+
+// Dice Roll functions
+var getDiceRoll = function () {
+  var randomNum = Math.floor(Math.random() * 6) + 1;
+  return randomNum;
+};
+
+// Concatenate the two dice roll numbers
+var combineTwoDiceRolls = function (diceRoll1, diceRoll2) {
+  var combinedDiceRolls = Number(String(diceRoll1) + String(diceRoll2));
+  return combinedDiceRolls;
+};
+
+// Compare functions
+var compareCombinedNumber = function (firstNum, secondNum) {
+  // var outcome = `No. of P1 wins:${numOfPlayer1Wins}; No.of P2 wins: ${numOfPlayer2Wins}.<br><br>`;
+  if (firstNum > secondNum) {
+    numOfPlayer1Wins += 1;
+    outcome = `No. of P1 wins: ${numOfPlayer1Wins} <br> No. of P2 wins: ${numOfPlayer2Wins}<br><br> Player 1 wins! His number of ${firstNum} is bigger!`;
+  } else if (firstNum < secondNum) {
+    numOfPlayer2Wins += 1;
+    outcome = `No. of P1 wins: ${numOfPlayer1Wins} <br> No. of P2 wins: ${numOfPlayer2Wins}<br><br> Player 2 wins! His number of ${secondNum} is bigger!`;
+  }
+  return outcome;
+};
+
+// Waiting for Player one mode
+var playerOneMode = function () {
+  // player 1 roll dice
+  player1diceRoll1 = getDiceRoll();
+  console.log(`Roll 1:${player1diceRoll1}`);
+  player1diceRoll2 = getDiceRoll();
+  console.log(`Roll 2:${player1diceRoll2}`);
+  return (myOutputValue = `🎲 WELCOME PLAYER 1 🎲 <br><br> You rolled ${player1diceRoll1} for dice one, ${player1diceRoll2} for dice two. <br><br> Choose the order of the dice by entering "1" or "2"`);
+};
+
+// Player one mode
+var playerOneChooseMode = function (input) {
```

Comment:
consider refactoring `playerOneChooseMode` and `playerTwoChooseMode` into a single function